site/app/controllers/admin/PlagiarismController.php

```diff
@@ -214,105 +261,108 @@ public function showPlagiarismResult($gradeable_id) {
 
+
+        // Version
+        $version_option = $_POST['version_option'];
+        assert($version_option == "active_version" || $version_option == "all_versions");
+
+
+        // Regex
+        // TODO: Can we find a way to validate the regex here to tell the user their regex was invalid before feeding it to Lichen?
+        assert(isset($_POST["regex_dir"]) && isset($_POST["regex_to_select_files"]));
+        $regex_directories = $_POST["regex_dir"];
+        $regex_for_selecting_files = $_POST['regex_to_select_files'];
+
+
+        // Language
+        assert(isset($_POST['language']));
         $language = $_POST['language'];
-        if (isset($_POST['threshold']) && $_POST['threshold'] !== '' && is_numeric($_POST['threshold']) && $_POST['threshold'] >= 0) {
+        if (!in_array($language, PlagiarismUtils::getSupportedLanguages())) {
+            $this->core->addErrorMessage("Invalid selected language");
+            $this->core->redirect($return_url);
+        }
+
+
+        // Common code threshold
+        if (isset($_POST['threshold']) && $_POST['threshold'] !== '' && is_numeric($_POST['threshold']) && $_POST['threshold'] >= 2) {
             $threshold = $_POST['threshold'];
         }
         else {
             $this->core->addErrorMessage("Invalid input provided for threshold");
             $this->core->redirect($return_url);
         }
 
-        if (isset($_POST['sequence_length']) && $_POST['sequence_length'] !== '' && is_numeric($_POST['sequence_length']) && $_POST['sequence_length'] >= 2) {
+
+        // Sequence length
+        if (isset($_POST['sequence_length']) && $_POST['sequence_length'] !== '' && is_numeric($_POST['sequence_length']) && $_POST['sequence_length'] >= 1) {
             $sequence_length = $_POST['sequence_length'];
         }
         else {
             $this->core->addErrorMessage("Invalid input provided for sequence length. The minimum allowed threshold value is 2");
             $this->core->redirect($return_url);
         }
 
+
+        // Prior terms
+        $prev_gradeable_number = $_POST['prior_term_gradeables_number'];
         $prev_term_gradeables = [];
         for ($i = 0; $i < $prev_gradeable_number; $i++) {
             if ($_POST['prev_sem_' . $i] != "" && $_POST['prev_course_' . $i] != "" && $_POST['prev_gradeable_' . $i] != "") {
-                array_push($prev_term_gradeables, "/var/local/submitty/course/" . $_POST['prev_sem_' . $i] . "/" . $_POST['prev_course_' . $i] . "/submissions/" . $_POST['prev_gradeable_' . $i]);
+                array_push($prev_term_gradeables, FileUtils::joinPaths($course_path, $_POST['prev_sem_' . $i], $_POST['prev_course_' . $i], "submissions", $_POST['prev_gradeable_' . $i]));
             }
         }
 
-        $ignore_submissions = [];
+        // Submissions to ignore
         $ignore_submission_option = $_POST['ignore_submission_option'];
+        assert($ignore_submission_option == "ignore" || $ignore_submission_option == "no_ignore");
```

Comment:
I wasn't sure how to properly handle this.  There can only ever be two values (`ignore` and `no_ignore`) and the frontend should know that.  If the value isn't one of those things, it's the programmer's fault, not the user's fault and this should theoretically never happen so I thought that throwing a PHP error would be appropriate.  Suggestions on what the correct thing to do here is?  If it isn't one of these two values, my thought was that we want to kill the script here before it goes on with the invalid data and starts corrupting things.

Reply:
Better would probably be to add an error message and just kick the user back to the form. We want to avoid having uncaught exceptions in the system as it's a poor UX for the end-user to commit an action and then see "An exception has occurred" screen, versus an error banner that states `Invalid ignore submission options, expected "ignore" or "no_ignore", got "<foo>".` Or something. A future improvement would be also saving state within the form when they're kicked back to it, but that can be a future thing.

Reply:
Ok, will update.